src/object.d

```diff
@@ -111,9 +111,9 @@ int __cmp(T)(scope const T[] lhs, scope const T[] rhs) @trusted
     }
 }
 
-// Compare class and interface objects for ordering.
-private int __cmp(Obj)(Obj lhs, Obj rhs)
-if (is(Obj : Object))
+// Compare class objects for ordering.
+int __cmp(C1, C2)(C1 lhs, C2 rhs)
+if (is(C1 == class) && is(C2 == class))
```

Comment:
Can you use `if ((is(C1 == class) || is(C1 == typeof(null))) && (is(C2 == class) || is(C2 == typeof(null))))` to handle the null condition here and avoid modifying the unittests?